packages/client/tests/functional/json-null-types/tests.ts

```diff
@@ -28,14 +29,19 @@ testMatrix.setupTestSuite(
     })
 
     describe('requiredJsonField', () => {
-      test('JsonNull', async () => {
-        const data = await prisma.requiredJsonField.create({
-          data: {
-            json: Prisma.JsonNull,
-          },
-        })
-        expect(data.json).toBe(null)
-      })
+      // TODO adapter does not seem to make a difference between JsonNull and DbNull
+      // Error converting field "json" of expected non-nullable type "Json", found incompatible value of "null".
```

Comment:
Looks like something we need to investigate

Reply:
Quick reproduction.

## Prisma Schema

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgres"
  url      = env("JS_PG_DATABASE_URL")
}

model NullableJsonField {
  id Int @id @default(autoincrement())
  json Json?
}

model RequiredJsonField {
  id Int @id @default(autoincrement())
  json Json
}
```

## TypeScript

```
import { Pool } from 'pg'
import { PrismaClient, Prisma } from '@prisma/client'

async function main() {
  const db = new Pool({
    connectionString: process.env.JS_PG_DATABASE_URL as string,
  })

  const prisma = new PrismaClient()

  await prisma.requiredJsonField.deleteMany()

  await prisma.requiredJsonField.create({
    data: {
      json: Prisma.JsonNull,
    },
  })

  const prismaResult = await prisma.requiredJsonField.findFirst()

  // { ..., json: null }
  console.log('prismaResult', prismaResult)

  // oid: null
  // rows: [ { json: null } ],
  // fields: [ Field { dataTypeID: 3802 } ]
  // Note: type ID 3802 represents "JSONB"
  const pgResult = await db.query('SELECT "json" FROM "public"."RequiredJsonField"')
  console.log('pgResult', pgResult)
}

main()
```